### PR TITLE
use privileged loopback instead of raw registries

### PR DESCRIPTION
**Pull request by deads2k**
Most of the added lines are generated.  I can split it out if necessary.

This stops using raw registries for our user and identity storage in favor of using privileged loopback connections.  This will stop us from skipping authorization and admission checks when using these resources.

@openshift/sig-security @enj @simo5

### CODE REVIEW

**Comment by deads2k** [comment]
types.go update is for generated clients. approved

File: pkg/cmd/server/kubernetes/master/master_config_test.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ var expectedGroupPreferredVersions []string = []string{
 	"rbac.authorization.k8s.io/v1beta1",
 	"settings.k8s.io/v1alpha1",
 	"storage.k8s.io/v1",
+	"user.openshift.io/v1",
```
Comment (enj) [inline pkg/cmd/server/kubernetes/master/master_config_test.go]: wat?

File: pkg/user/registry/identity/strategy.go [hunk 2]
```diff
@@ -45,6 +45,11 @@ func (identityStrategy) PrepareForCreate(ctx apirequest.Context, obj runtime.Obj
 	identity.Name = identityName(identity.ProviderName, identity.ProviderUserName)
 }
 
+func identityName(provider, identity string) string {
+	// TODO: normalize?
+	return provider + ":" + identity
```
Comment (enj) [inline pkg/user/registry/identity/strategy.go]: This probably deserves a doc saying it cannot change?

File: pkg/auth/userregistry/identitymapper/lookup_test.go [hunk 3]
```diff
@@ -154,3 +162,83 @@ func TestLookup(t *testing.T) {
...
+	Create(ctx apirequest.Context, obj runtime.Object, _ bool) (runtime.Object, error)
+	Update(ctx apirequest.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error)
+}
+
+// storage puts strong typing around storage calls
+type storage struct {
+	mappingStorage
+}
+
+// NewRegistry returns a new Registry interface for the given Storage. Any mismatched
```
Comment (enj) [inline pkg/auth/userregistry/identitymapper/lookup_test.go]: Is all of this registry code in this test just to make the old interface based tests work?

File: pkg/cmd/server/kubernetes/master/master_config_test.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/server/kubernetes/master/master_config_test.go]: It's based on an imported install.  This package ended up picking it up by accident when the client poked them.

File: pkg/user/registry/identity/strategy.go (hunk 2 above)
Comment (deads2k) [inline pkg/user/registry/identity/strategy.go]: ok

File: pkg/auth/userregistry/identitymapper/lookup_test.go (hunk 3 above)
Comment (deads2k) [inline pkg/auth/userregistry/identitymapper/lookup_test.go]: > Is all of this registry code in this test just to make the old interface based tests work?

Yeah.  I'll let the testing debt be cleaned up separately.  The interfaces in the real code are now correct and the existing testing registries could be removed or they could be left.  I have no preference.

**Comment by deads2k** [comment]
comments addressed.

**Pull request merged**